src/controllers/attachments/index.ts

```diff
@@ -1,32 +1,120 @@
-import { Controller, Get, Route } from 'tsoa'
+import { UploadedFile, Request, Controller, Get, Route, Path, Post, Header, Response } from 'tsoa'
 import { Logger } from 'pino'
+import express from 'express'
 
 import { logger } from '../../lib/logger'
-import Database from '../../lib/db'
+import Database, { Models, Query } from '../../lib/db'
 import type { Attachments } from '../../models'
+import { BadRequst, NotFound } from '../../lib/error-handler'
+
+/**
+ * A successfull reponse
+ * @example { "status": 200, "data": [ { "id": "5a3f54bd-a459-45fb-8e50-9ffdd26ff981", "filename": "Screenshot 2023-03-13 at 15.26.41.png", "created_at": "2023-03-14T11:56:19.832Z" } ] }
+ */
+interface Success {
+  status: 200 | 201
+  message?: string
+  data?: Attachments[] | Attachments
+  headers?: { [k: string]: unknown }
+}
+
+type File = {
+  [k: string]: string
+}
 
 @Route('attachments')
 export class attachments extends Controller {
   log: Logger
-  // TMP update once we have more defined schema
-  dbClient: any = new Database()
-  db: any
+  dbClient: Database = new Database()
+  db: Models<Query> | any // TMP this is the only one could not address now
 
   constructor() {
     super()
     this.log = logger.child({ controller: '/attachments' })
-    this.db = this.dbClient.db()
+    // TMP will be updated with a wrapper so ORM client independant
+    this.db = this.dbClient.init()
   }
 
   @Get('/')
-  public async get(): Promise<{ status: number; attachments: Attachments[] }> {
-    this.log.debug({
-      msg: 'this is a test route to validate tsoa/swagger',
-    })
+  public async get(): Promise<Success> {
+    this.log.debug('retrieving all attachments')
+    const result = await this.db.attachments()
 
     return {
       status: 200,
-      attachments: await this.db.attachments(),
+      data: result.map(({ binary_blob, ...item }: Attachments) => ({
+        ...item,
+        size: binary_blob.size,
+      })),
     }
   }
+
+  @Post('/')
+  public async create(@Request() req: express.Request, @UploadedFile() file: File): Promise<Success> {
+    this.log.debug(`crating an attachment ${JSON.stringify(file || req.body)}`)
+    if (file) {
+      await this.db.attachments().insert({
+        filename: file.originalname,
+        binary_blob: Buffer.from(file.buffer),
+      })
+      return {
+        status: 201,
+        message: 'created as file blob',
+      }
```

Comment:
do we need to include `status codes in the body?

Reply:
yes, TSOA then forward it. You can use `@SuccessResponse()` decorator so in this instance you can say

```ts
@SuccessResponse<Success>
.... (req: Request...
```

Reply:
I think `@SuccessResponse` can be used without needing `status` in the body though? e.g. the following code returns a `201`

```
  @Post()
  @SuccessResponse('201')
  public async createCapacity(@Body() requestBody: DemandRequest): Promise<DemandResponse> {

   ...db logic

    return {
      id: capacity.id,
      owner: ownerAlias,
      status: capacity.status,
      parametersAttachmentId: parametersAttachmentId,
    }
  }
```

Reply:
then `response.body` doesn't contain `status` or `message` is can just be `'created as file blob'`

Reply:
Yes, indeed was doubling up on statuses. I've removed and unwrapped from data object since it's a body either way.